### ISSUE TITLE
Support BDD style keywords and given/when/then/but prefixes

### ISSUE THREAD

**Issue by robaca** [jivesoftware/robot-intellij-plugin#42]
Currently the editor doesn't support BDD style keywords that have variable placeholders inside the keyword. 
These keywords that are usually also prefixed with given/when/then/but which is also not understood by the editor

**Comment by charlescapps** [comment]
I added support for keywords with variables in the name a few months ago. Are you sure you have the newest version of the plugin? Try going to IntelliJ -> Check for updates, and make sure you have version 1.7.1.

As to prefixing with given/when/then/but, can you explain what feature would be needed here with a concrete example? You can certainly add that inside Robot comments with the # symbol, I'm not sure what else you are looking for?

**Comment by robaca** [comment]
Yes, you are right. If I remove the prefixes, the keyword usage is detected correctly.

I'm using the plugin with version 1.7.1 in IntelliJ IDEA Ultimate 13.1.6. I checked out the [webdemo](https://bitbucket.org/robotframework/webdemo/wiki/Home) and renamed the .txt files to .robot. 

Keywords in test cases like this are not recognized by the plugin:

``` robot
    Given browser is opened to login page
    When user "demo" logs in with password "mode"
    Then welcome page should be open
```

What I would expect:
- the usage of the keywords ("browser is opened to login page", ...) would be detected so that Ctrl-(mouseover|mouseclick|B) and "Find Usages" work
- the prefixes given/when/then/and/but should get their own highlighting to visually separate them from the "real" keyword

The optimum would be to also support BDD style test cases written in other languages like german that have their own prefixes. For that we created a "German-BDD.robot" in our project containing keywords like this:

``` robot
Gegeben sei ${keyword}
    Run Keyword  ${keyword}

Es existiert ${keyword}
    Run Keyword  ${keyword}

...
```

Possible solutions that come into my mind:
- reading a configuration text file containing additional prefixes from inside the project (so that it can be checked into scm)
- parsing the keywords and detecting parameters that themselves expect a keyword (by parameter name "keyword" or some other mechanism)

There also exist other keywords that expect a keyword passed as parameters (like "Run Keyword And Ignore Error",  table syntax). Maybe a more general solution is possible.

**Comment by charlescapps** [comment]
I'll take a look, I may have just not allowed quotes in the lexer for Robot
Keywords. If quotes are allowed by Robot, that would be a bug I should fix.

It's probably as simple as removing quotes to form the Java method, just as
Robot removes underscores and whitespace.

I'll fix this when I get a chance, I'm working on another project, but I
could probably take a look this weekend!

~Charles Capps
